packages/blockchain-wallet-v4-frontend/src/modals/Btc/VerifyMessage/index.js

```diff
@@ -0,0 +1,142 @@
+// Verify the signature of a message signed with a Bitcoin address.
+
+import {
+  Banner,
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  TooltipIcon,
+  TooltipHost
+} from 'blockchain-info-components'
+
+import { FormItem, FormLabel, TextArea, TextBox } from 'components/Form'
+import * as core from './core'
+import modalEnhancer from 'providers/ModalEnhancer'
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+import { validBitcoinAddress } from 'services/FormHelper'
+
+const item = (label, input) => (
+  <FormItem style={{ marginBottom: `1rem` }}>
+    <FormLabel>
+      <div style={{ marginBottom: `0.5rem` }}>{label}</div>
+      {input}
+    </FormLabel>
+  </FormItem>
+)
+
+const ItemAddress = ({ address, network, onChange }) =>
+  item(
+    <FormattedMessage
+      id='modals.verifyMessage.address'
+      defaultMessage='Bitcoin Address:'
+    />,
+    <TextBox
+      input={{
+        onChange,
+        name: 'address'
+      }}
+      meta={{
+        error: validBitcoinAddress(address, null, { network }),
+        touched: address !== ``
+      }}
+    />
+  )
+
+const ItemMessage = ({ onChange }) =>
+  item(
+    <FormattedMessage
+      id='modals.verifyMessage.message'
+      defaultMessage='Message:'
+    />,
+    <TextArea
+      input={{
+        name: 'message',
+        onChange
+      }}
+      meta={{}}
+    />
+  )
+
+const ItemSignature = ({ onChange }) =>
+  item(
+    <FormattedMessage
+      id='modals.verifyMessage.signature'
+      defaultMessage='Signature:'
+    />,
+    <TextArea
+      input={{
+        name: 'signature',
+        onChange
+      }}
+      meta={{}}
+    />
+  )
+
+class VerifyMessage extends React.PureComponent {
+  constructor (props) {
+    super(props)
+    this.state = { address: ``, message: ``, signature: `` }
+  }
+
+  render () {
+    const { close, network } = this.props
+
+    const onChange = ({ target: { name, value } }) => {
```

Comment:
creating functions in `render` causes perf issues, it's always best to declare them as a method and use `this.onChange`

Reply:
Good catch!